src/components/elements/tableFilters/filters/FilterMenu.tsx

```diff
@@ -28,8 +28,8 @@ const TableFilterMenu = <T,>(props: TableFilterMenuProps<T>): JSX.Element => {
       .filter(([k]) => props.filters.hasOwnProperty(k))
       // Count # of filters that have values applied
       .filter(([, v]) => (Array.isArray(v) ? !isEmpty(v) : !isNil(v)))
-      // Convert to human-readable strings
-      .map(([k]) => startCase(k));
+      // Get the human-readable label of this filter
+      .map(([k]) => props.filters[k as keyof T]?.label);
```

Comment:
Maybe `|| startCase(k)` since it looks like could be undefined?

Reply:
Good plan! It shouldn't be undefined because of the filter by `hasOwnProperty` above, but agree, just in case